coding-guides/raw-events.md

```diff
@@ -40,30 +40,31 @@ Let's add a `raw` event listener. Usually the given items are called a packet, s
 
 ```javascript
 client.on('raw', packet => {
-    // We don't want this to run on unrelated packets
-    if (!['MESSAGE_REACTION_ADD', 'MESSAGE_REACTION_REMOVE'].includes(packet.t)) return;
-    // Grab the channel to check the message from
-    const channel = client.channels.get(packet.d.channel_id);
-    // There's no need to emit if the message is cached, because the event will fire anyway for that
-    if (channel.messages.has(packet.d.message_id)) return;
-    // Since we have confirmed the message is not cached, let's fetch it
-    channel.fetchMessage(packet.d.message_id).then(message => {
-        // Emojis can have identifiers of name:id format, so we have to account for that case as well
-        const emoji = packet.d.emoji.id ? `${packet.d.emoji.name}:${packet.d.emoji.id}` : packet.d.emoji.name;
-        // This gives us the reaction we need to emit the event properly, in top of the message object
-        const reaction = message.reactions.get(emoji);
-        // Adds the currently reacting user to the reaction's users collection.
-        if (reaction) reaction.users.set(packet.d.user_id, client.users.get(packet.d.user_id));
-        // Check which type of event it is before emitting
-        if (packet.t === 'MESSAGE_REACTION_ADD') {
-            client.emit('messageReactionAdd', reaction, client.users.get(packet.d.user_id));
-        }
-        if (packet.t === 'MESSAGE_REACTION_REMOVE') {
-            client.emit('messageReactionRemove', reaction, client.users.get(packet.d.user_id));
-        }
-    });
-});
+   // We don't want this to run on unrelated packets
+   if (!['MESSAGE_REACTION_ADD', 'MESSAGE_REACTION_REMOVE'].includes(packet.t)) { return; }
+   // Grab the channel to check the message from
+   const channel = client.channels.resolve(packet.d.channel_id);
+   // There's no need to emit if the message is cached, because the event will fire anyway for that
+   if (channel.messages.cache.get(packet.d.message_id)) { return; }
+   // Since we have confirmed the message is not cached, let's fetch it
+   channel.messages.fetch(packet.d.message_id).then(function (message) {
```

Comment:
I agree with the change from `fetchMessage` to `messages.fetch` but why change the function... it's perfectly valid.

Reply:
https://discord.js.org/#/docs/main/stable/class/Channel  again v12 breaking changes ... pretty sure `fetchMessage` doesnt exist anymore

Reply:
York's point was that changing from a valid arrow function to using a function keyword isn't useful. Please revere this change to `message => {`, as it's the way it's supposed to be.

Reply:
If you wish to revert it go for it.  I have washed my hands of v12 and the breaking changes their of.  I simply wished to put a way forward for anyone who stumbles into this same quagmire in the future.